mods/pageviews.js

```diff
@@ -320,9 +305,25 @@ PJVS.prototype.pageviewsForProjects = function(restbase, req) {
             }
         }
 
-    });
+    }).catch(notFoundCatcher);
 
-    return dataRequest.then(normalizeResponse);
+    return dataRequest.then(normalizeResponse).then(function(res) {
+        if (res.body.items) {
+            res.body.items.forEach(function(item) {
+                // prefer the v column if it's loaded
+                if (item.hasOwnProperty('v') && item.v !== null) {
```

Comment:
Not true, if (item.v) fails for v === 0

On Wed, Nov 4, 2015 at 12:15 PM, Marko Obrovac <notifications@github.com>
wrote:

> In mods/pageviews.js
> <https://github.com/wikimedia/restbase/pull/396#discussion_r43909215>:
>
> >
> > -    return dataRequest.then(normalizeResponse);
> > +    return dataRequest.then(normalizeResponse).then(function(res) {
> > +        if (res.body.items) {
> > +            res.body.items.forEach(function(item) {
> > +                // prefer the v column if it's loaded
> > +                if (item.hasOwnProperty('v') && item.v !== null) {
>
> if (item.v) should be enough given that by the time this code is
> deployed, the underlying table will be updated.
>
> Also, it might be a good idea to run a script that goes through all of the
> records and moves the views values into v, so that such checks may be
> removed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wikimedia/restbase/pull/396/files#r43909215>.
>